tsfc/fiatinterface.py

```diff
@@ -76,6 +76,88 @@
 have a direct FIAT equivalent."""
 
 
+class FlattenToQuad(FIAT.FiniteElement):
+    """A wrapper class that flattens a FIAT quadrilateral element defined
+    on a TensorProductCell to one with FiredrakeQuadrilateral entities
+    and tabulation properties."""
+
+    def __init__(self, element):
+        """ Constructs a FlattenToQuad element.
+
+        :arg element: a fiat element
+        """
+        self.element = element
+        nodes = element.dual.nodes
+        self.ref_el = FiredrakeQuadrilateral()
+        entity_ids = element.dual.entity_ids
+        flat_entity_ids = {}
+        flat_entity_ids[0] = entity_ids[(0, 0)]
+        flat_entity_ids[1] = dict(enumerate(entity_ids[(0, 1)].values() +
+                                            entity_ids[(1, 0)].values()))
```

Comment:
I recently spotted that even though this has been working very well for a while, it is actually wrong. Asking the `.values()` of a `dict` returns the values in an unspecified order. Since the keys are `range(n)` for a small `n`, by pure implementation luck this returns the ordering that we need.